mt-kahypar/partition/refinement/flows/parallel_construction.cpp

```diff
@@ -40,9 +40,15 @@ template<typename GraphAndGainTypes>
 typename ParallelConstruction<GraphAndGainTypes>::TmpHyperedge
 ParallelConstruction<GraphAndGainTypes>::DynamicIdenticalNetDetection::get(const size_t he_hash,
                                                                         const vec<whfc::Node>& pins) {
+  if constexpr (PartitionedHypergraph::is_graph) {
+    // there can't be identical nets for graphs
+    return TmpHyperedge { 0, std::numeric_limits<size_t>::max(), whfc::invalidHyperedge };
+  }
+
   const size_t bucket_idx = he_hash % _hash_buckets.size();
   if ( __atomic_load_n(&_hash_buckets[bucket_idx].threshold, __ATOMIC_RELAXED) == _threshold ) {
     // There exists already some hyperedges with the same hash
+    _hash_buckets[bucket_idx].lock.lock();
```

Comment:
Is this a place to use a lockguard (releases lock when the object is destroyed)?

Reply:
I reverted the changes to the construction as they are not the focus of the PR and, it seems, also didn't result in an improvement